### PR TITLE
ARXIVNG-2965: update base release, fix wsgi

### PR DESCRIPTION
**Pull request by mhl10**
This updates the base release to 0.16.6rc8, primarily to support configurable web analytics. Also includes some fixes to the wsgi scripts comparable to the fixes we introduced in arxiv-browse. I pinned the version of `werkzeug` to an older version for now, mainly to maintain compatibility with arxiv-auth.

I'll do a squash merge on this once accepted.

### CODE REVIEW

File: wsgi.py [hunk 1]
```diff
@@ -7,21 +7,30 @@
...
 logger = logging.getLogger(__name__)
 
-__flask_app__ = create_ui_web_app()
+__flask_app__ = None
 
 
 def application(environ, start_response):
     """WSGI application factory."""
     for key, value in environ.items():
+        # Copy string WSGI envrion to os.environ. This is to get apache
```
Comment (bdc34) [inline wsgi.py]: I sort of wonder if we should abandon using apache SetEnv vars to configure flask.

Some other options are env vars on the process or a python config file.

File: search/controllers/advanced/tests.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 from unittest import TestCase, mock
 from datetime import date, datetime
 from dateutil.relativedelta import relativedelta
-from werkzeug import MultiDict
+from werkzeug.datastructures import MultiDict
```
Comment (bdc34) [inline search/controllers/advanced/tests.py]: I wonder if we can get rid of our direct dependence on werkzeug by not using its data structures?

File: wsgi.py (hunk 1 above)
Comment (DavidLFielding) [inline wsgi.py]: Minor: envrion probably should be environ.

Comment (mhl10) [reply wsgi.py]: I'd be favorable moving towards python config files, since SetEnv vars have been finicky and error prone.

**Comment by mhl10** [comment]
> I am not sure how to test these changes. When I run either this branch or develop using this command:
> FLASK_APP=app.py FLASK_DEBUG=1 ELASTICSEARCH_SERVICE_HOST=127.0.0.1 pipenv run flask run
> 
> I get this error in the browser:
> flask.cli.NoAppException
> 
> And this error in the terminal:
> ModuleNotFoundError: No module named 'flask_s3'

Try doing a `pipenv sync` before you run the flask app. Since these are fairly minor changes, you don't even need to spin up the Elasticsearch backend if you don't want to, so you can just run `FLASK_APP=app.py FLASK_DEBUG=1 pipenv run flask run` and then access http://localhost:5000/. When you view the source, there should be no Matomo config.

File: search/controllers/advanced/tests.py (hunk 2 above)
Comment (mhl10) [reply search/controllers/advanced/tests.py]: There is a separate `multidict` package but I think there is some convenience in having werkzeug bundle a bunch of the HTTP-friendly data structures together.

**Pull request merged**